src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,316 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
+
+
+/**
+ * The parser for Duke.
+ */
+public class Parser {
+
+    /**
+     * Returns a string array from the parsed user input string.
+     * @param rawInput The user input.
+     * @return String array from the parsed input [COMMAND, arguments...].
+     * @throws DukeException An invalid user input will produce this exception.
+     */
+    public String[] parseInput(String rawInput) throws DukeException {
+        String[] inputs = rawInput.split("\\s+");
+        if (inputs.length < 1) {
+            throw new DukeException(DukeException.Errors.INVALID_COMMAND.toString());
+        }
+        Constant.Command command;
+        String commandStr = inputs[0].toUpperCase();
+        try {
+            command = Constant.Command.valueOf(commandStr);
+        } catch (Exception e) {
+            throw new DukeException(DukeException.Errors.INVALID_COMMAND.toString());
+        }
+        switch (command) {
+        case LIST:
+            if (inputs.length != 1) {
+                throw new DukeException(DukeException.Errors.INVALID_ARGUMENT.toString()
+                        + " `list` command has no arguments");
+            }
+            return new String[] {commandStr};
+
+        case DONE:
+            if (inputs.length != 2) {
+                throw new DukeException(DukeException.Errors.INVALID_ARGUMENT.toString()
+                        + " (example: 'done 5')");
+            }
+            try {
+                // Checks if the argument is a number.
+                Integer.parseInt(inputs[1]);
+            } catch (Exception e) {
+                throw new DukeException(DukeException.Errors.WRONG_ARGUMENT_TYPE.toString()
+                        + " (example: 'done 5')");
+            }
+            return new String[] {commandStr, inputs[1]};
+
+        case TODO:
+            if (inputs.length < 2) {
+                throw new DukeException(DukeException.Errors.MISSING_DESCRIPTION.toString()
+                        + " (example: 'todo watch Borat')");
+            }
+            String description = combineStringArray(inputs, 1, inputs.length);
+            return new String[] {commandStr, description};
+
+        case DEADLINE:
+            if (inputs.length < 2) {
+                throw new DukeException(DukeException.Errors.MISSING_DESCRIPTION.toString() + " (example: 'deadline watch Borat /by 2021-08-21 18:00')");
```

Comment:
I like the use of an example string to guide the user to a correct format for each command, but perhaps this line should be broken up because it is too long (over 120 characters)